### PR TITLE
fix(api): Delete should require team:admin instead of team:write

### PR DESCRIPTION
**Pull request by schew2381**
The DELETE should require `team:admin` over `team:write` per our [scopes documentation](https://docs.sentry.io/api/permissions/#teams).

There is a bug in team-roles where the function `can_admin_team` that we use a lot only checks for `team:write`. Team admins also have access to `team:admin` so we should include a check for this

### CODE REVIEW

File: src/sentry/api/endpoints/organization_member/__init__.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def can_admin_team(access: Access, team: Team) -> bool:
         return True
     if not access.has_team_membership(team):
         return False
-    return access.has_team_scope(team, "team:write")
+    return access.has_team_scope(team, "team:write") or access.has_team_scope(team, "team:admin")
```
Comment (schew2381) [inline src/sentry/api/endpoints/organization_member/__init__.py:77]: Ideally `has_team_scope` would take in a list of scopes to reduce the amount of calls, but that is out of scope for this PR.

File: src/sentry/api/endpoints/organization_member/team_details.py [hunk 2]
```diff
@@ -83,7 +85,7 @@ def _is_org_owner_or_manager(access: Access) -> bool:
 
 @region_silo_endpoint
 class OrganizationMemberTeamDetailsEndpoint(OrganizationMemberEndpoint):
-    permission_classes = [TeamOrgMemberPermission]
+    permission_classes = [OrganizationTeamMemberPermission]
```
Comment (EricHasegawa) [inline src/sentry/api/endpoints/organization_member/team_details.py:88]: these are usually tuples using round brackets - is there a reason for the square ones?

File: tests/sentry/api/endpoints/test_organization_member_team_details.py [hunk 3]
```diff
@@ -581,7 +583,7 @@ def test_manager_can_remove_members(self):
     def test_manager_can_remove_members_using_user_token(self):
         self.login_as(self.manager)
 
-        scopes = ["org:write", "team:write"]
+        scopes = ["org:write", "team:admin"]
```
Comment (EricHasegawa) [inline tests/sentry/api/endpoints/test_organization_member_team_details.py:586]: should we check if this works using _only_ the `team:admin` scope?

Comment (schew2381) [reply tests/sentry/api/endpoints/test_organization_member_team_details.py:586]: The test checks this because its looping over each scope individually

File: src/sentry/api/endpoints/organization_member/team_details.py (hunk 2 above)
Comment (schew2381) [reply src/sentry/api/endpoints/organization_member/team_details.py:88]: I think it was like and I didn't notice, I'll change it to a tuple

File: src/sentry/api/endpoints/organization_member/team_details.py [hunk 4]
```diff
@@ -83,7 +85,7 @@ def _is_org_owner_or_manager(access: Access) -> bool:
 
 @region_silo_endpoint
 class OrganizationMemberTeamDetailsEndpoint(OrganizationMemberEndpoint):
-    permission_classes = [TeamOrgMemberPermission]
+    permission_classes = OrganizationTeamMemberPermission
```
Comment (cathteng) [inline src/sentry/api/endpoints/organization_member/team_details.py:88]:
```suggestion
    permission_classes = (OrganizationTeamMemberPermission,)
```

File: tests/sentry/api/endpoints/test_organization_member_team_details.py [hunk 5]
```diff
@@ -489,15 +490,16 @@ def test_admin_cannot_remove_member(self):
     def test_admin_cannot_remove_member_using_user_token(self):
         # admin not in team
         self.login_as(self.admin)
-        token = self.create_user_auth_token(user=self.admin_user, scope_list=["team:write"])
-        self.get_error_response(
+        token = self.create_user_auth_token(user=self.admin_user, scope_list=["team:admin"])
```
Comment (cathteng) [inline tests/sentry/api/endpoints/test_organization_member_team_details.py:493]: this is because the user token `team:admin` scope is `team:admin` on the org level?

**Review by cathteng — APPROVED** [review]
lgtm, but make sure OrganizationTeamMemberPermission is inside a tuple

File: tests/sentry/api/endpoints/test_organization_member_team_details.py (hunk 5 above)
Comment (schew2381) [reply tests/sentry/api/endpoints/test_organization_member_team_details.py:493]: Org admins have `team:admin`, but they shouldn't be able to act on teams they're not a member of
